gh-109350: Fix outdated captured output in unittest.mock documentation

ragulpr opened the pull request:
Fixes https://github.com/python/cpython/issues/109350

Updates the captured output/exceptions that are not reflecting the current (more readable) exceptions ([Doc/library/unittest.mock.rst](https://github.com/python/cpython/blob/5dcbbd8861e618488d95416dee8ea94577e3f4f0/Doc/library/unittest.mock.rst), [Doc/library/unittest.mock-examples.rst](https://github.com/python/cpython/blob/5dcbbd8861e618488d95416dee8ea94577e3f4f0/Doc/library/unittest.mock-examples.rst)) . 

I ran this @ Python 3.13.0a0, I'm sure it can be backported far back but I'm not up to the task right now.

<!-- gh-issue-number: gh-109350 -->
* Issue: gh-109350
<!-- /gh-issue-number -->


<!-- readthedocs-preview cpython-previews start -->
----
:books: Documentation preview :books:: https://cpython-previews--109353.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

ragulpr committed:
Update outdated docs output

ragulpr commented on Doc/library/unittest.mock.rst:
```diff
@@ -957,7 +959,7 @@ object::
         Traceback (most recent call last):
         ...
-        AssertionError: Expected mock to have been awaited once. Awaited 2 times.
+        AssertionError: Expected method to have been awaited once. Awaited 0 times.
```
This is the actual but likely not the intended output, if called with `mock.assert_awaited_once()` rather than `mock.method.assert_awaited_once()` it yields the output that was documented. Should I change it immediately or spawn separate issue?

cjw296 replied:
Fix it here :-)

ragulpr replied:
Awesome! Done [d7f3b13](https://github.com/python/cpython/pull/109353/commits/d7f3b135cfe21bfda530901877188b8a623a6429) ✅

ragulpr committed:
Merge branch 'main' into unittest-mock-docs

ragulpr committed:
Merge branch 'main' into unittest-mock-docs

cjw296 reviewed (approved):
Fantastic, thanks for this!

ragulpr committed:
mock.method.assert_awaited_once()=>mock.assert_awaited_once()
Run with
```
import asyncio
import unittest
from unittest.mock import Mock, MagicMock, AsyncMock, patch, call, sentinel

mock = AsyncMock()
async def main():
    await mock()

asyncio.run(main())
mock.assert_awaited_once()
asyncio.run(main())
mock.assert_awaited_once()
```

ragulpr committed:
Merge branch 'main' into unittest-mock-docs
